### PR TITLE
fishPlugins.fzf-fish: init at 5.6

### PR DESCRIPTION
**Pull request by pacien**
###### Motivation for this change

This adds a package for the fzf.fish fish plugin.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/shells/fish/plugins/clownfish.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  version = "unstable-2021-01-17";
+
+  src = fetchFromGitHub {
+    owner = "IlanCosman";
+    repo = "clownfish";
+    rev = "a0db28d8280d05561b8f48c0465480725feeca4c";
+    sha256 = "04xvikyrdm6yxh588vbpwvm04fas76pa7sigsaqrip7yh021xxab";
+  };
+
+  meta = {
```
[the remark this answers is not in the document]
Comment (pacien) [reply pkgs/shells/fish/plugins/clownfish.nix]: Amended.

File: pkgs/shells/fish/plugins/default.nix [hunk 2]
```diff
@@ -4,10 +4,17 @@ lib.makeScope newScope (self: with self; {
 
   buildFishPlugin = callPackage ./build-fish-plugin.nix { };
 
+  clownfish = callPackage ./clownfish.nix { };
+
+  # Fishtape 2.x and 3.x aren't compatible,
+  # but both versions are used in the tests of different other plugins.
   fishtape = callPackage ./fishtape.nix { };
+  fishtape3 = callPackage ./fishtape3.nix { };
```
[the remark this answers is not in the document]
Comment (pacien) [reply pkgs/shells/fish/plugins/default.nix]: Amended.

File: pkgs/shells/fish/plugins/fishtape3.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  checkPhase = ''
+    fishtape tests/*.fish
+  '';
+
+  meta = {
+    description = "100% pure-Fish test runner";
+    homepage = "https://github.com/jorgebucaran/fishtape";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ pacien ];
+  };
```
[the remark this answers is not in the document]
Comment (pacien) [reply pkgs/shells/fish/plugins/fishtape3.nix]: Amended.

File: pkgs/shells/fish/plugins/fzf-fish.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    rm -r tests/search_shell_variables
+
+    fishtape tests/*/*.fish
+  '';
+
+  meta = {
+    description = "Augment your fish command line with fzf key bindings";
+    homepage = "https://github.com/PatrickF1/fzf.fish";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ pacien ];
```
[the remark this answers is not in the document]
Comment (pacien) [reply pkgs/shells/fish/plugins/fzf-fish.nix]: Amended.

**Comment by pacien** [comment]
@SuperSandro2000 Thank you for your review; I amended your suggestions.

CC: @teehemkay (I saw that you were trying to package this plugin as well)

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 115519` at 74a66734 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/c857bf0de06e311339773b069210a0e4">fishPlugins.clownfish</a></li></li>
    <li><a href="https://gist.github.com/c59768b13dbfc9ffa312fb978cacf37a">fishPlugins.fishtape_3</a></li></li>
    <li><a href="https://gist.github.com/4c0b0ccaef1ecd7024a4c9ab0231462e">fishPlugins.fzf-fish</a></li></li>
  </ul>
</details>

**Comment by teehemkay** [comment]
> @SuperSandro2000 Thank you for your review; I amended your suggestions.
> 
> CC: @teehemkay (I saw that you were trying to package this plugin as well)

I was indeed, but didn't managed to do it successfully.

I was having this weird issue where the loop that executes the scripts in `conf.d` in `wrapper.nix` was failing silently thus breaking the plugins...
But if I ran the script manually it worked...

This was with a test environment setup using `nix-shell` and a `fishWrapper` and I was packaging the missing plugins using `niv`.

Had to give up looking into it due to lack of time (and understanding ;-)
But will circle back to it at a later time (maybe this week-end). There's a couple of other plugins I'd like to package as well (`z`, `autopair` and `tide`)

Also waiting for `fish` 3.2 to land in `unstable`

Meanwhile this looks fine to me. Thank you for packaging those!

Thanks

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 115519` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>fishPlugins.clownfish</li>
<li>fishPlugins.fishtape_3</li>
<li>fishPlugins.fzf-fish</li>
</ul>
</details>

**Comment by baracoder** [comment]
@teehemkay did you make any progress on packaging the `tide` plugin?

**Comment by teehemkay** [comment]
> @teehemkay did you make any progress on packaging the `tide` plugin?

Nope

**Pull request merged**